### ISSUE TITLE
[gatsby-plugin-offline] I have a finite (and small) number of pages in my app. Can I force the plugin to pre-cache these pages?

### ISSUE THREAD

**Issue by evanwinter** [gatsbyjs/gatsby#15802]
## Summary

I have a small app that has several pages, all found at `src/pages/`:
* index.js
* welcome.js
* products.js
* recipes.js
* contact.js
* quiz.js
* quiz-results.js

Ideally, I would like all of these pages to be precached, so that a user can navigate to them when offline, without having to first visit them while online.

Currently, I'm getting the following behavior -- if I visit the page while online, it gets cached. I can then go offline, and still view this page. If I don't visit the page while online, it doesn't get cached. When I navigate to it, I get the offline error page.

Is there a way to ensure that gatsby-plugin-offline precaches all of my pages? 

Here is what it precaches now:
![Screen Shot 2019-07-16 at 3 25 03 PM](https://user-images.githubusercontent.com/19195468/61327483-a2f42b80-a7de-11e9-8d34-e19615b84a0f.png)

Does that seem like an expected list of pre-cached assets based on what I've described above? Or should there be more in there?

Thank you in advance for any help!

### Environment (if relevant)

```
gatsby info --clipboard

  System:
    OS: macOS 10.14.5
    CPU: (4) x64 Intel(R) Core(TM) i5-4570 CPU @ 3.20GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 11.12.0 - /usr/local/bin/node
    Yarn: 1.15.2 - /usr/local/bin/yarn
    npm: 6.9.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  Browsers:
    Chrome: 75.0.3770.142
    Firefox: 66.0.3
    Safari: 12.1.1
  npmPackages:
    gatsby: ^2.3.34 => 2.3.34
    gatsby-image: ^2.0.41 => 2.0.41
    gatsby-plugin-google-analytics: ^2.0.19 => 2.0.19
    gatsby-plugin-layout: ^1.0.15 => 1.0.15
    gatsby-plugin-manifest: ^2.0.29 => 2.0.29
    gatsby-plugin-offline: ^2.2.4 => 2.2.4
    gatsby-plugin-react-helmet: ^3.0.12 => 3.0.12
    gatsby-plugin-react-svg: ^2.1.1 => 2.1.1
    gatsby-plugin-sass: ^2.0.11 => 2.0.11
    gatsby-plugin-sharp: ^2.0.35 => 2.0.35
    gatsby-source-contentful: ^2.0.54 => 2.0.54
    gatsby-source-filesystem: ^2.0.33 => 2.0.33
    gatsby-transformer-sharp: ^2.1.18 => 2.1.18
  npmGlobalPackages:
    gatsby-cli: 2.5.4
```

### File contents (if changed)

`gatsby-config.js`: N/A 
`package.json`: N/A 
`gatsby-node.js`: N/A 
`gatsby-browser.js`: N/A 
`gatsby-ssr.js`: N/A

**Comment by evanwinter** [comment]
To clarify, the end result I want is that the entire app is cached on first load, without having to visit each page.

I'm trying the approach I found here: https://github.com/gatsbyjs/gatsby/issues/9383

Hopefully that will resolve my issue. I will close this out if not!

**Comment by evanwinter** [comment]
Okay, I figured out that I probably need to change the `globPatterns` Workbox config option, since the plugin is only doing the app shell and whatever it deems to be "critical assets".  I tried doing "*/**", as someone in that other thread did, and that (obviously) was far more greedy than I intended, precaching over 700 files (including multiple versions of every image). 

This sort of worked, except for I started to get some "Loading chunk [n] failed" when I tried to navigate around. I assume that has something to do with the fact that the cache is now pretty overloaded... I'm now trying to figure out if I can be more selective. 

If there is any advice out there that may help me, I would love to hear it! 

Thank you in advance. 👨🏻‍💻
